[Regression] Note content assertion in `test_pull_handles_uncommitted_submodule_commits()` fails on Windows

This is on branch `fix-resource-permission-error-diff2-args`. This probably needs to be debugged on a physical Windows box. Iteration is too slow for `ssh`-ing onto CI runner instances. See below.

```python
================================== FAILURES ===================================
_______________ test_pull_handles_uncommitted_submodule_commits _______________

tmp_path = WindowsPath('C:/Users/runneradmin/AppData/Local/Temp/pytest-of-runneradmin/pytest-0/test_pull_handles_uncommitted_0')

    def test_pull_handles_uncommitted_submodule_commits(tmp_path: Path):
        UNCOMMITTED_SM: SampleCollection = get_test_collection(
            "uncommitted_submodule_commits"
        )
        UNCOMMITTED_SM_EDITED: SampleCollection = get_test_collection(
            "uncommitted_submodule_commits_edited"
        )
        runner = CliRunner()
        japanese_gitrepo_path = Path(JAPANESE_GITREPO_PATH).resolve()
        with runner.isolated_filesystem(temp_dir=tmp_path):
    
            JAPANESE_SUBMODULE_DIRNAME = "japanese-core-2000"
    
            # Clone collection.
            out = clone(runner, UNCOMMITTED_SM.col_file)
            logger.debug(f"\n{out}")
    
            # Check that the content of a note in the collection is correct.
            os.chdir(UNCOMMITTED_SM.repodir)
            with open(
                Path(JAPANESE_SUBMODULE_DIRNAME) / "\u305d\u308c.md", "r", encoding="UTF-8"
            ) as f:
                note_text = f.read()
                expected = "that, that one\nthat, that one\nthis, this one"
                logger.debug(f"SM note text:\n{note_text}")
                assert expected in note_text
            os.chdir("../")
    
            # Delete `japanese-core-[200](https://github.com/langfield/ki/runs/7827544817?check_suite_focus=true#step:9:201)0/` subdirectory, and commit.
            sm_dir = Path(UNCOMMITTED_SM.repodir) / JAPANESE_SUBMODULE_DIRNAME
            F.rmtree(F.test(sm_dir))
            repo = git.Repo(UNCOMMITTED_SM.repodir)
            repo.git.add(all=True)
            repo.index.commit("Delete cloned `japanese-core-2000` folder.")
            repo.close()
    
            # Push the deletion.
            os.chdir(UNCOMMITTED_SM.repodir)
            out = push(runner)
            logger.debug(out)
    
            # Copy a new directory of notes to `japanese-core-2000/` subdirectory,
            # and initialize it as a git repository.
            submodule_name = JAPANESE_SUBMODULE_DIRNAME
            shutil.copytree(japanese_gitrepo_path, submodule_name)
            git.Repo.init(submodule_name, initial_branch=BRANCH_NAME)
            sm = git.Repo(submodule_name)
            sm.git.add(all=True)
            _ = sm.index.commit("Initial commit.")
    
            # Add as a submodule.
            repo.git.submodule("add", Path(submodule_name).resolve())
            repo.git.add(all=True)
            _ = repo.index.commit("Add submodule.")
            repo.close()
    
            # Push changes.
            out = push(runner)
            logger.debug(out)
    
            # Add a new line to a note, and commit the addition in the submodule.
            with open(
                Path(JAPANESE_SUBMODULE_DIRNAME) / "\u305d\u308c.md", "a", encoding="UTF-8"
            ) as f:
                f.write("A new line at the bottom.")
            sm.git.add(all=True)
            _ = sm.index.commit("Added a new line.")
            sm.close()
    
            # Edit collection.
            shutil.copyfile(UNCOMMITTED_SM_EDITED.col_file, UNCOMMITTED_SM.col_file)
    
            # Pull changes from collection to root ki repository.
            out = pull(runner)
            logger.debug(out)
            assert "fatal: remote error: " not in out
    
            with open(
                Path(JAPANESE_SUBMODULE_DIRNAME) / "\u305d\u308c.md", "r", encoding="UTF-8"
            ) as f:
                note_text = f.read()
            logger.debug(f"SM note text:\n{note_text}")
            expected_mackerel = "\nholy mackerel\n"
            expected_this = "\nthis, this one\n"
>           assert expected_mackerel in note_text
E           assert '\\nholy mackerel\\n' in '## Note\\nnid: 166049[207](https://github.com/langfield/ki/runs/7827544817?check_suite_focus=true#step:9:208)0243\\nmodel: iKnow! Vocabulary\\ntags: \\nmarkdown: false\\n\\n### Expression\\n\u305d\u308c\\n\\n### Meaning\\nthat, that one\\nthat, that one\\n\\n### Reading\\n\\n\\n### Audio\\n[sound:88de14505716052e8fc8b61d809d2843.mp3]\\n\\n### Image_URI\\n<img src="fefcf8b5fc45942801a9a5c4645[228](https://github.com/langfield/ki/runs/7827544817?check_suite_focus=true#step:9:229)a0.jpg">\\n\\n### iKnowID\\nitem:435851\\n\\n### iKnowType\\nitem\\n'

tests\test_integration.py:995: AssertionError
```

--- a/ki/__init__.py
+++ b/ki/__init__.py
@@ -2185,8 +2185,10 @@ def _pull(kirepo: KiRepo, silent: bool) -> None:
                 patch_path: ExtantFile = F.write(patch_path, patch.diff.text)
                 if sys.platform == "win32":
                     with open(patch_path, "rb") as f:
-                        with open(patch_path, "wb") as g:
-                            g.write(f.read().replace(b"\r\n", b"\n"))
+                        patch_bytes = f.read()
+                    reformatted_bytes = patch_bytes.replace(b"\r\n", b"\n")
+                    with open(patch_path, "wb") as f:
+                        f.write(reformatted_bytes)
 
                 # Number of leading path components to drop from diff paths.
                 num_parts = len(sm_rel_root.parts) + 1

--- a/tests/test_integration.py
+++ b/tests/test_integration.py
@@ -93,7 +93,6 @@ EDITED: SampleCollection = get_test_collection("edited")
 # CLI
 
 
-@pytest.mark.skip
 def test_bad_command_is_bad():
     """Typos should result in errors."""
     result = invoke(ki.ki, ["clome"])
@@ -101,7 +100,6 @@ def test_bad_command_is_bad():
     assert "Error: No such command 'clome'." in result.output
 
 
-@pytest.mark.skip
 def test_runas_module():
     """Can this package be run as a Python module?"""
     command = "python -m ki --help"
@@ -109,14 +107,12 @@ def test_runas_module():
     assert completed.returncode == 0
 
 
-@pytest.mark.skip
 def test_entrypoint():
     """Is entrypoint script installed? (setup.py)"""
     result = invoke(ki.ki, ["--help"])
     assert result.exit_code == 0
 
 
-@pytest.mark.skip
 def test_version():
     """Does --version display information as expected?"""
     expected_version = version("ki")
@@ -126,7 +122,6 @@ def test_version():
     assert result.exit_code == 0
 
 
-@pytest.mark.skip
 def test_command_availability():
     """Are commands available?"""
     results = []
@@ -137,7 +132,6 @@ def test_command_availability():
         assert result.exit_code == 0
 
 
-@pytest.mark.skip
 def test_cli():
     """Does CLI stop execution w/o a command argument?"""
     with pytest.raises(SystemExit):
@@ -149,7 +143,6 @@ def test_cli():
 
 
 @beartype
-@pytest.mark.skip
 def test_fails_without_ki_subdirectory(tmp_path: Path):
     """Do pull and push know whether they're in a ki-generated git repo?"""
     runner = CliRunner()
@@ -164,7 +157,6 @@ def test_fails_without_ki_subdirectory(tmp_path: Path):
 
 
 @beartype
-@pytest.mark.skip
 def test_computes_and_stores_md5sum(tmp_path: Path):
     """Does ki add new hash to `.ki/hashes`?"""
     ORIGINAL: SampleCollection = get_test_collection("original")
@@ -200,7 +192,6 @@ def test_computes_and_stores_md5sum(tmp_path: Path):
             assert f"199216c39eeabe23a1da016a99ffd3e2  {ORIGINAL.filename}" in hashes
 
 
-@pytest.mark.skip
 def test_no_op_pull_push_cycle_is_idempotent():
     """Do pull/push not misbehave if you keep doing both?"""
     ORIGINAL: SampleCollection = get_test_collection("original")
@@ -226,7 +217,6 @@ def test_no_op_pull_push_cycle_is_idempotent():
         push(runner)
 
 
-@pytest.mark.skip
 def test_output(tmp_path: Path):
     """Does it print nice things?"""
     ORIGINAL: SampleCollection = get_test_collection("original")
@@ -263,7 +253,6 @@ def test_output(tmp_path: Path):
 # CLONE
 
 
-@pytest.mark.skip
 def test_clone_fails_if_collection_doesnt_exist():
     """Does ki clone only if `.anki2` file exists?"""
     ORIGINAL: SampleCollection = get_test_collection("original")
@@ -277,7 +266,6 @@ def test_clone_fails_if_collection_doesnt_exist():
         assert not os.path.isdir(ORIGINAL.repodir)
 
 
-@pytest.mark.skip
 def test_clone_fails_if_collection_is_already_open():
     """Does ki print a nice error message when Anki is accidentally left open?"""
     ORIGINAL: SampleCollection = get_test_collection("original")
@@ -289,7 +277,6 @@ def test_clone_fails_if_collection_is_already_open():
             clone(runner, ORIGINAL.col_file)
 
 
-@pytest.mark.skip
 def test_clone_creates_directory():
     """Does it create the directory?"""
     ORIGINAL: SampleCollection = get_test_collection("original")
@@ -303,7 +290,6 @@ def test_clone_creates_directory():
 
 
 @beartype
-@pytest.mark.skip
 def test_clone_displays_errors_from_creation_of_kirepo_metadata(mocker: MockerFixture):
     """Do errors get displayed nicely?"""
     ORIGINAL: SampleCollection = get_test_collection("original")
@@ -319,7 +305,6 @@ def test_clone_displays_errors_from_creation_of_kirepo_metadata(mocker: MockerFi
 
 
 @beartype
-@pytest.mark.skip
 def test_clone_displays_errors_from_loading_kirepo_at_end(mocker: MockerFixture):
     """Do errors get propagated in the places we expect?"""
     ORIGINAL: SampleCollection = get_test_collection("original")
@@ -370,7 +355,6 @@ def test_clone_displays_errors_from_loading_kirepo_at_end(mocker: MockerFixture)
             clone(runner, ORIGINAL.col_file)
 
 
-@pytest.mark.skip
 def test_clone_handles_html():
     """Does it tidy html and stuff?"""
     HTML: SampleCollection = get_test_collection("html")
@@ -389,7 +373,6 @@ def test_clone_handles_html():
         assert snippet in contents
 
 
-@pytest.mark.skip
 def test_clone_tidying_only_breaks_lines_for_fields_containing_html():
     """Does it tidy html and stuff?"""
     HTML: SampleCollection = get_test_collection("html")
@@ -410,7 +393,6 @@ def test_clone_tidying_only_breaks_lines_for_fields_containing_html():
         )
 
 
-@pytest.mark.skip
 def test_clone_errors_when_directory_is_populated():
     """Does it disallow overwrites?"""
     ORIGINAL: SampleCollection = get_test_collection("original")
@@ -429,7 +411,6 @@ def test_clone_errors_when_directory_is_populated():
             clone(runner, ORIGINAL.col_file)
 
 
-@pytest.mark.skip
 def test_clone_cleans_up_on_error():
     """Does it clean up on nontrivial errors?"""
     HTML: SampleCollection = get_test_collection("html")
@@ -449,7 +430,6 @@ def test_clone_cleans_up_on_error():
             os.environ["PATH"] = old_path
 
 
-@pytest.mark.skip
 def test_clone_cleans_up_preserves_directories_that_exist_a_priori():
     """Does clone not delete targetdirs that already existed?"""
     HTML: SampleCollection = get_test_collection("html")
@@ -469,7 +449,6 @@ def test_clone_cleans_up_preserves_directories_that_exist_a_priori():
             os.environ["PATH"] = old_path
 
 
-@pytest.mark.skip
 def test_clone_displays_nice_errors_for_missing_dependencies():
     """Does it tell the user what to install?"""
     HTML: SampleCollection = get_test_collection("html")
@@ -511,7 +490,6 @@ def test_clone_displays_nice_errors_for_missing_dependencies():
             os.environ["PATH"] = old_path
 
 
-@pytest.mark.skip
 def test_clone_succeeds_when_directory_exists_but_is_empty():
     """Does it clone into empty directories?"""
     ORIGINAL: SampleCollection = get_test_collection("original")
@@ -523,7 +501,6 @@ def test_clone_succeeds_when_directory_exists_but_is_empty():
         clone(runner, ORIGINAL.col_file)
 
 
-@pytest.mark.skip
 def test_clone_generates_expected_notes():
     """Do generated note files match content of an example collection?"""
     ORIGINAL: SampleCollection = get_test_collection("original")
@@ -545,7 +522,6 @@ def test_clone_generates_expected_notes():
         assert cloned_md5 == true_md5
 
 
-@pytest.mark.skip
 def test_clone_generates_deck_tree_correctly():
     """Does generated FS tree match example collection?"""
     MULTIDECK: SampleCollection = get_test_collection("multideck")
@@ -569,7 +545,6 @@ def test_clone_generates_deck_tree_correctly():
         assert cloned_md5 == true_md5
 
 
-@pytest.mark.skip
 def test_clone_generates_ki_subdirectory():
     """Does clone command generate .ki/ directory?"""
     ORIGINAL: SampleCollection = get_test_collection("original")
@@ -584,7 +559,6 @@ def test_clone_generates_ki_subdirectory():
         assert os.path.isdir(kidir)
 
 
-@pytest.mark.skip
 def test_cloned_collection_is_git_repository():
     """Does clone run `git init` and stuff?"""
     ORIGINAL: SampleCollection = get_test_collection("original")
@@ -597,7 +571,6 @@ def test_cloned_collection_is_git_repository():
         assert is_git_repo(ORIGINAL.repodir)
 
 
-@pytest.mark.skip
 def test_clone_commits_directory_contents():
     """Does clone leave user with an up-to-date repo?"""
     ORIGINAL: SampleCollection = get_test_collection("original")
@@ -619,7 +592,6 @@ def test_clone_commits_directory_contents():
         assert len(commits) == 1
 
 
-@pytest.mark.skip
 def test_clone_leaves_collection_file_unchanged():
     """Does clone leave the collection alone?"""
     ORIGINAL: SampleCollection = get_test_collection("original")
@@ -634,7 +606,6 @@ def test_clone_leaves_collection_file_unchanged():
         assert original_md5 == updated_md5
 
 
-@pytest.mark.skip
 def test_clone_directory_argument_works():
     """Does clone obey the target directory argument?"""
     ORIGINAL: SampleCollection = get_test_collection("original")
@@ -651,7 +622,6 @@ def test_clone_directory_argument_works():
         assert os.path.isdir(target)
 
 
-@pytest.mark.skip
 def test_clone_writes_media_files():
     """Does clone copy media files from the media directory into 'MEDIA'?"""
     MEDIACOL: SampleCollection = get_test_collection("media")
@@ -664,7 +634,6 @@ def test_clone_writes_media_files():
         assert audio_path.is_file()
 
 
-@pytest.mark.skip
 def test_clone_handles_cards_from_a_single_note_in_distinct_decks(tmp_path: Path):
     SPLIT: SampleCollection = get_test_collection("split")
     runner = CliRunner()
@@ -674,7 +643,6 @@ def test_clone_handles_cards_from_a_single_note_in_distinct_decks(tmp_path: Path
         assert os.path.isfile(Path(SPLIT.repodir) / "top" / "a" / "a.md")
 
 
-@pytest.mark.skip
 def test_clone_url_decodes_media_src_attributes(tmp_path: Path):
     DOUBLE: SampleCollection = get_test_collection("no_double_encodings")
     runner = CliRunner()
@@ -691,7 +659,6 @@ def test_clone_url_decodes_media_src_attributes(tmp_path: Path):
 # PULL
 
 
-@pytest.mark.skip
 def test_pull_fails_if_collection_no_longer_exists():
     """Does ki pull only if `.anki2` file exists?"""
     ORIGINAL: SampleCollection = get_test_collection("original")
@@ -708,7 +675,6 @@ def test_pull_fails_if_collection_no_longer_exists():
             pull(runner)
 
 
-@pytest.mark.skip
 def test_pull_fails_if_collection_file_is_corrupted():
     """Does `pull()` fail gracefully when the collection file is bad?"""
     ORIGINAL: SampleCollection = get_test_collection("original")
@@ -726,7 +692,6 @@ def test_pull_fails_if_collection_file_is_corrupted():
             pull(runner)
 
 
-@pytest.mark.skip
 def test_pull_writes_changes_correctly():
     """Does ki get the changes from modified collection file?"""
     ORIGINAL: SampleCollection = get_test_collection("original")
@@ -746,7 +711,6 @@ def test_pull_writes_changes_correctly():
         assert os.path.isfile(NOTE_1)
 
 
-@pytest.mark.skip
 def test_pull_unchanged_collection_is_no_op():
     """Does ki remove remote before quitting?"""
     ORIGINAL: SampleCollection = get_test_collection("original")
@@ -766,7 +730,6 @@ def test_pull_unchanged_collection_is_no_op():
         assert orig_hash == new_hash
 
 
-@pytest.mark.skip
 def test_pull_avoids_unnecessary_merge_conflicts():
     """Does ki prevent gratuitous merge conflicts?"""
     ORIGINAL: SampleCollection = get_test_collection("original")
@@ -786,7 +749,6 @@ def test_pull_avoids_unnecessary_merge_conflicts():
         assert "Automatic merge failed; fix" not in out
 
 
-@pytest.mark.skip
 def test_pull_still_works_from_subdirectories():
     """Does pull still work if you're farther down in the directory tree than the repo route?"""
     ORIGINAL: SampleCollection = get_test_collection("original")
@@ -805,7 +767,6 @@ def test_pull_still_works_from_subdirectories():
         pull(runner)
 
 
-@pytest.mark.skip
 def test_pull_displays_errors_from_repo_ref():
     """Does 'pull()' return early when the last push commit ref is bad?"""
     ORIGINAL: SampleCollection = get_test_collection("original")
@@ -826,7 +787,6 @@ def test_pull_displays_errors_from_repo_ref():
             pull(runner)
 
 
-@pytest.mark.skip
 def test_pull_displays_errors_from_clone_helper(mocker: MockerFixture):
     ORIGINAL: SampleCollection = get_test_collection("original")
     runner = CliRunner()
@@ -847,7 +807,6 @@ def test_pull_displays_errors_from_clone_helper(mocker: MockerFixture):
             pull(runner)
 
 
-@pytest.mark.skip
 def test_pull_handles_unexpectedly_changed_checksums(mocker: MockerFixture):
     ORIGINAL: SampleCollection = get_test_collection("original")
     runner = CliRunner()
@@ -866,7 +825,6 @@ def test_pull_handles_unexpectedly_changed_checksums(mocker: MockerFixture):
             pull(runner)
 
 
-@pytest.mark.skip
 def test_pull_displays_errors_from_repo_initialization(mocker: MockerFixture):
     ORIGINAL: SampleCollection = get_test_collection("original")
     runner = CliRunner()
@@ -888,7 +846,6 @@ def test_pull_displays_errors_from_repo_initialization(mocker: MockerFixture):
 
 
 @pytest.mark.xfail
-@pytest.mark.skip
 def test_pull_preserves_reassigned_note_ids(tmp_path: Path):
     """UNFINISHED!"""
     ORIGINAL: SampleCollection = get_test_collection("original")
@@ -919,7 +876,6 @@ def test_pull_preserves_reassigned_note_ids(tmp_path: Path):
         raise NotImplementedError
 
 
-@pytest.mark.skip
 def test_pull_handles_non_standard_submodule_branch_names(tmp_path: Path):
     ORIGINAL: SampleCollection = get_test_collection("original")
     runner = CliRunner()
@@ -1032,7 +988,6 @@ def test_pull_handles_uncommitted_submodule_commits(tmp_path: Path):
         assert expected_this in note_text
 
 
-@pytest.mark.skip
 def test_pull_removes_files_deleted_in_remote(tmp_path: Path):
     ORIGINAL: SampleCollection = get_test_collection("original")
     DELETED: SampleCollection = get_test_collection("deleted")
@@ -1049,7 +1004,6 @@ def test_pull_removes_files_deleted_in_remote(tmp_path: Path):
         out = pull(runner)
 
 
-@pytest.mark.skip
 def test_pull_does_not_duplicate_decks_converted_to_subdecks_of_new_top_level_decks(
     tmp_path: Path,
 ):
@@ -1077,7 +1031,6 @@ def test_pull_does_not_duplicate_decks_converted_to_subdecks_of_new_top_level_de
 # PUSH
 
 
-@pytest.mark.skip
 def test_push_writes_changes_correctly(tmp_path: Path):
     """If there are committed changes, does push change the collection file?"""
     ORIGINAL: SampleCollection = get_test_collection("original")
@@ -1132,7 +1085,6 @@ def test_push_writes_changes_correctly(tmp_path: Path):
         assert len(new_notes) == 2
 
 
-@pytest.mark.skip
 def test_push_verifies_md5sum():
     """Does ki only push if md5sum matches last pull?"""
     ORIGINAL: SampleCollection = get_test_collection("original")
@@ -1151,7 +1103,6 @@ def test_push_verifies_md5sum():
             push(runner)
 
 
-@pytest.mark.skip
 def test_push_generates_correct_backup():
     """Does push store a backup identical to old collection file?"""
     ORIGINAL: SampleCollection = get_test_collection("original")
@@ -1187,7 +1138,6 @@ def test_push_generates_correct_backup():
         assert backup_exists
 
 
-@pytest.mark.skip
 def test_push_doesnt_write_uncommitted_changes():
     """Does push only write changes that have been committed?"""
     ORIGINAL: SampleCollection = get_test_collection("original")
@@ -1209,7 +1159,6 @@ def test_push_doesnt_write_uncommitted_changes():
         assert len(os.listdir(".ki/backups")) == 0
 
 
-@pytest.mark.skip
 def test_push_doesnt_fail_after_pull():
     """Does push work if we pull and then edit and then push?"""
     ORIGINAL: SampleCollection = get_test_collection("original")
@@ -1249,7 +1198,6 @@ def test_push_doesnt_fail_after_pull():
         push(runner)
 
 
-@pytest.mark.skip
 def test_no_op_push_is_idempotent():
     """Does push not misbehave if you keep pushing?"""
     ORIGINAL: SampleCollection = get_test_collection("original")
@@ -1269,7 +1217,6 @@ def test_no_op_push_is_idempotent():
         push(runner)
 
 
-@pytest.mark.skip
 def test_push_deletes_notes():
     """Does push remove deleted notes from collection?"""
     ORIGINAL: SampleCollection = get_test_collection("original")
@@ -1300,7 +1247,6 @@ def test_push_deletes_notes():
         assert not os.path.isfile(NOTE_0)
 
 
-@pytest.mark.skip
 def test_push_still_works_from_subdirectories():
     """Does push still work if you're farther down in the directory tree than the repo route?"""
     ORIGINAL: SampleCollection = get_test_collection("original")
@@ -1326,7 +1272,6 @@ def test_push_still_works_from_subdirectories():
         push(runner)
 
 
-@pytest.mark.skip
 def test_push_deletes_added_notes():
     """Does push remove deleted notes added with ki?"""
     ORIGINAL: SampleCollection = get_test_collection("original")
@@ -1384,7 +1329,6 @@ def test_push_deletes_added_notes():
 # We expect this test not to work anymore, because it is now outside the scope
 # of `ki`.
 @pytest.mark.xfail
-@pytest.mark.skip
 def test_push_generates_correct_title_for_notes():
     """Does push use the truncated sort field as a filename?"""
     ORIGINAL: SampleCollection = get_test_collection("original")
@@ -1416,7 +1360,6 @@ def test_push_generates_correct_title_for_notes():
         assert "r.md" in notes
 
 
-@pytest.mark.skip
 def test_push_displays_informative_error_when_last_push_file_is_missing():
     ORIGINAL: SampleCollection = get_test_collection("original")
     runner = CliRunner()
@@ -1435,7 +1378,6 @@ def test_push_displays_informative_error_when_last_push_file_is_missing():
             push(runner)
 
 
-@pytest.mark.skip
 def test_push_honors_ignore_patterns():
     ORIGINAL: SampleCollection = get_test_collection("original")
     runner = CliRunner()
@@ -1471,7 +1413,6 @@ def test_push_honors_ignore_patterns():
         assert "Warning: not Anki note" in out
 
 
-@pytest.mark.skip
 def test_push_displays_errors_from_head_ref_maybes(mocker: MockerFixture):
     ORIGINAL: SampleCollection = get_test_collection("original")
     runner = CliRunner()
@@ -1493,7 +1434,6 @@ def test_push_displays_errors_from_head_ref_maybes(mocker: MockerFixture):
             push(runner)
 
 
-@pytest.mark.skip
 def test_push_displays_errors_from_head_repo_ref(mocker: MockerFixture):
     ORIGINAL: SampleCollection = get_test_collection("original")
     runner = CliRunner()
@@ -1517,7 +1457,6 @@ def test_push_displays_errors_from_head_repo_ref(mocker: MockerFixture):
             push(runner)
 
 
-@pytest.mark.skip
 def test_push_displays_errors_from_notetype_parsing_in_push_deltas_during_model_adding(
     mocker: MockerFixture,
 ):
@@ -1549,7 +1488,6 @@ def test_push_displays_errors_from_notetype_parsing_in_push_deltas_during_model_
             push(runner)
 
 
-@pytest.mark.skip
 def test_push_displays_errors_from_notetype_parsing_in_push_deltas_during_push_flatnote_to_anki(
     mocker: MockerFixture,
 ):
@@ -1582,7 +1520,6 @@ def test_push_displays_errors_from_notetype_parsing_in_push_deltas_during_push_f
             out = push(runner)
 
 
-@pytest.mark.skip
 def test_push_handles_submodules(tmp_path: Path):
     ORIGINAL: SampleCollection = get_test_collection("original")
     runner = CliRunner()
@@ -1614,7 +1551,6 @@ def test_push_handles_submodules(tmp_path: Path):
         assert "<br>z<br>" in notes[0]["Back"]
 
 
-@pytest.mark.skip
 def test_push_writes_media(tmp_path: Path):
     MEDIACOL: SampleCollection = get_test_collection("media")
     runner = CliRunner()
@@ -1656,7 +1592,6 @@ def test_push_writes_media(tmp_path: Path):
         assert len(check.unused) == 0
 
 
-@pytest.mark.skip
 def test_push_handles_foreign_models(tmp_path: Path):
     """Just check that we don't return an exception from `push()`."""
     ORIGINAL: SampleCollection = get_test_collection("original")
@@ -1672,7 +1607,6 @@ def test_push_handles_foreign_models(tmp_path: Path):
         out = push(runner)
 
 
-@pytest.mark.skip
 def test_push_fails_if_database_is_locked():
     """Does ki print a nice error message when Anki is accidentally left open?"""
     ORIGINAL: SampleCollection = get_test_collection("original")
@@ -1692,7 +1626,6 @@ def test_push_fails_if_database_is_locked():
             push(runner)
 
 
-@pytest.mark.skip
 def test_push_is_nontrivial_when_pulled_changes_are_reverted(tmp_path: Path):
     """
     If you push, make changes in Anki, then pull those changes, then undo them
@@ -1748,7 +1681,6 @@ def test_push_is_nontrivial_when_pulled_changes_are_reverted(tmp_path: Path):
         assert "ki push: up to date." not in out
 
 
-@pytest.mark.skip
 def test_push_doesnt_unnecessarily_deduplicate_notetypes():
     """
     Does push refrain from adding a new notetype if the requested notetype
@@ -1804,7 +1736,6 @@ def test_push_doesnt_unnecessarily_deduplicate_notetypes():
         col.close(save=False)
 
 
-@pytest.mark.skip
 def test_push_is_nontrivial_when_pushed_changes_are_reverted_in_repository():
     """
     The following operation should be nontrivial:
@@ -1850,7 +1781,6 @@ def test_push_is_nontrivial_when_pushed_changes_are_reverted_in_repository():
         assert "ki push: up to date." not in out
 
 
-@pytest.mark.skip
 def test_push_changes_deck_for_moved_notes():
     MULTIDECK: SampleCollection = get_test_collection("multideck")
     runner = CliRunner()
@@ -1885,7 +1815,6 @@ def test_push_changes_deck_for_moved_notes():
         assert colnote.deck == "aa::dd"
 
 
-@pytest.mark.skip
 def test_push_is_trivial_for_committed_submodule_contents(tmp_path: Path):
     UNCOMMITTED_SM: SampleCollection = get_test_collection(
         "uncommitted_submodule_commits"
@@ -1926,7 +1855,6 @@ def test_push_is_trivial_for_committed_submodule_contents(tmp_path: Path):
         assert "ki push: up to date." in out
 
 
-@pytest.mark.skip
 def test_push_prints_informative_warning_on_push_when_subrepo_was_added_instead_of_submodule(
     tmp_path: Path,
 ):
@@ -1956,7 +1884,6 @@ def test_push_prints_informative_warning_on_push_when_subrepo_was_added_instead_
             assert "'git submodule add'" in out
 
 
-@pytest.mark.skip
 def test_push_handles_tags_containing_trailing_commas():
     COMMAS: SampleCollection = get_test_collection("commas")
     runner = CliRunner()
@@ -1980,7 +1907,6 @@ def test_push_handles_tags_containing_trailing_commas():
         push(runner)
 
 
-@pytest.mark.skip
 def test_push_correctly_encodes_quotes_in_html_tags(tmp_path: Path):
     BROKEN: SampleCollection = get_test_collection("broken_media_links")
     runner = CliRunner()

